pkg/component/autoscaling/vpa/vpa_test.go

```diff
@@ -1680,6 +1742,89 @@ var _ = Describe("VPA", func() {
 				managedResourceName = "shoot-core-vpa"
 			})
 
+			Context("Different ServiceMonitor labels", func() {
+				Context("should always attach `prometheus=shoot` to `admission-controller`", func() {
```

Comment:
Can't we use the same approach here as the above tests for the seed? E.g. first test both ServiceMonitors when `IsGardenCluster=true` then both when `IsGardenCluster=false`

Reply:
Thanks for the suggested improvement. You can find the restructured version of the `shoot` ServiceMonitor test cases in my latest updates.
Now both `vpa-admission-controller` and `vpa-recommeder` ServiceMonitors are being checked within the context of `IsGardenCluster` configuration - when `true` or `false` in separate contexts.